cache/disk/disk.go

```diff
@@ -134,10 +134,8 @@ func New(dir string, maxSizeBytes int64, storageMode string, proxy cache.Proxy)
 
 		f := filepath.Join(dir, c.FileLocation(kind, value.legacy, hash, value.size, value.random))
 
-		err := os.Remove(f)
-		if err != nil {
-			log.Printf("ERROR: failed to remove evicted cache file: %s", f)
-		}
+		// Run in a goroutine so we can release the lock sooner.
+		go removeFile(f)
```

Comment:
To avoid spawning a new goroutine every time, we could write the filename to a buffered channel, and have some goroutines reading from that channel and removing files.